### PR TITLE
Build commands

### PR DESCRIPTION
**Pull request by iwko**

### CODE REVIEW

File: biggerquery/cli.py [hunk 1]
```diff
@@ -417,6 +460,19 @@ def _cli_deploy_image(args):
                         vault_secret=args.vault_secret
                         )
 
+def _cli_build_image(args):
+    cmd = 'python setup.py build_project --build-image' + (' --e' if args.export_image_to_file else '')
```
Comment (beczkowb) [inline biggerquery/cli.py:464]: `project_setup.py` instead of `setup.py`

Comment (iwko) [reply biggerquery/cli.py]: Done

File: biggerquery/cli.py [hunk 2]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+    _create_build_dags_parser(subparsers)
+    _create_build_image_parser(subparsers)
+    _create_build_package_parser(subparsers)
+    _create_build_parser(subparsers)
+
     return parser.parse_args(args)
 
 
+def _create_build_parser(subparsers):
+    parser = subparsers.add_parser('build', description='Builds docker image, DAG files and .whl package from local files.')
```
Comment (bartoszwalacik) [inline biggerquery/cli.py:242]: a Docker image

Comment (bartoszwalacik) [inline biggerquery/cli.py:242]: from local source files ?

File: biggerquery/cli.py [hunk 3]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+    _add_build_dags_parser_arguments(parser)
+    _add_build_image_parser_arguments(parser)
+
+
+def _create_build_package_parser(subparsers):
+    subparsers.add_parser('build-package', description='Builds .whl package from local files.')
```
Comment (bartoszwalacik) [inline biggerquery/cli.py:248]: from local source files ?

File: biggerquery/cli.py [hunk 4]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+
+
+def _add_build_dags_parser_arguments(parser):
+    parser.add_argument('-w', '--workflow',
+                        type=str,
+                        help='The id of the workflow to start.')
```
Comment (bartoszwalacik) [inline biggerquery/cli.py:262]: An Id of a workflow to build ?

Comment (bartoszwalacik) [inline biggerquery/cli.py:262]: what's a worklfow Id, how can I get it?

File: biggerquery/cli.py [hunk 5]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+    parser.add_argument('-t', '--start-time',
+                        help='The date and time for which workflow should work.')
```
Comment (bartoszwalacik) [inline biggerquery/cli.py]: 'Point of time from which a workflow should start work.'
'For workflows triggered hourly -- datetime in format: ..., for example ...
'For workflows triggered daily -- date in format: ..., for example ...

File: biggerquery/cli.py [hunk 6]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+
+
+def _add_build_image_parser_arguments(parser):
+    parser.add_argument('-e', '--export-image-to-file',
+                        default=False,
+                        action="store_true",
+                        help="If set to true image will be exported to file.")
```
Comment (bartoszwalacik) [inline biggerquery/cli.py:272]: If true, an image is exported to a *.tar file.

File: biggerquery/cli.py [hunk 7]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+
+
+def _create_build_dags_parser(subparsers):
+    parser = subparsers.add_parser('build-dags',
+                                   description='Builds DAG files from local files.')
```
Comment (bartoszwalacik) [inline biggerquery/cli.py:277]: from local source files ?

File: biggerquery/cli.py [hunk 8]
```diff
@@ -229,9 +230,51 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+    _add_build_dags_parser_arguments(parser)
+
+
+def _create_build_image_parser(subparsers):
+    parser = subparsers.add_parser('build-image',
+                                   description='Builds docker image from local files.')
```
Comment (bartoszwalacik) [inline biggerquery/cli.py:283]: from local source files ?

File: biggerquery/cli.py (hunk 7 above)
Comment (bartoszwalacik) [inline biggerquery/cli.py:277]: Builds DAG files from local sources to {current_dir}/.dags

File: biggerquery/cli.py (hunk 5 above)
Comment (iwko) [reply biggerquery/cli.py]: I think for daily workflows we can pass datetime too

Comment (bartoszwalacik) [reply biggerquery/cli.py]: We need datetime examples here

Comment (bartoszwalacik) [reply biggerquery/cli.py]: hourly workflow 2020-07-01 10:00:00
daily workflow 2020-07-01

**Comment by bartoszwalacik** [comment]
We should improve/unify these two similar arguments:

run command help:
```
  -w WORKFLOW, --workflow WORKFLOW
                        The id of the workflow to start.
  -r RUNTIME, --runtime RUNTIME
                        The date and time when this job or workflow should be
                        started. The default is now (2020-08-06 08:39:38).
                        Examples: 2019-01-01, 2020-01-01 01:00:00
```

build-dags command help:
```
  -w WORKFLOW, --workflow WORKFLOW
                        Id of a workflow to build.
  -t START_TIME, --start-time START_TIME
                        Point of time from which a workflow should start
                        working.
```

**Comment by bartoszwalacik** [comment]
I have called `bgq build` and nothing happened. No exception handling?

```
(.venv) polpc05579:biggerquery bartosz.walacik$ python bgq build
/Users/bartosz.walacik/Tools/git_repo/biggerquery
(.venv) polpc05579:biggerquery bartosz.walacik$ 
```

File: biggerquery/cli.py (hunk 2 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

Comment (iwko) [reply biggerquery/cli.py]: Done

File: biggerquery/cli.py (hunk 3 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

File: biggerquery/cli.py (hunk 4 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

Comment (iwko) [reply biggerquery/cli.py]: See now

File: biggerquery/cli.py (hunk 5 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

File: biggerquery/cli.py (hunk 6 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

File: biggerquery/cli.py (hunk 8 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

File: biggerquery/cli.py (hunk 7 above)
Comment (iwko) [reply biggerquery/cli.py]: Done

**Pull request merged**